pkg/ldapsession/search.go

```diff
@@ -137,3 +170,17 @@ PagedSearch:
 	}
 	return nil
 }
+
+// ExecuteBulkSearchRequest will take a slice of ldap.SearchRequest and execute each one sequentially,
+// keeping the results channels open until the end of the last one
+func (w *LDAPSession) ExecuteBulkSearchRequest(searchRequests []*ldap.SearchRequest) error {
+	w.keepChannelsOpen()
+	for _, request := range searchRequests {
+		err := w.ExecuteSearchRequest(request)
+		if err != nil {
+			return err
+		}
+	}
+	w.CloseChannels()
```

Comment:
This one should be deferred, otherwise channels won't be closed in case an error occurs